pkg/apis/monitoring/v1/types.go

```diff
@@ -1603,6 +1603,25 @@ type AlertmanagerSpec struct {
 	// This is an alpha field and requires enabling StatefulSetMinReadySeconds feature gate.
 	// +optional
 	MinReadySeconds *uint32 `json:"minReadySeconds,omitempty"`
+	// GlobalAlertmanagerConfig used to set the global alertmanager config.
+	GlobalAlertmanagerConfig *GlobalAlertmanagerConfig `json:"globalAlertmanagerConfig,omitempty"`
+}
+
+type GlobalAlertmanagerConfig struct {
```

Comment:
Are we ok with adding and promoting these types straight into `v1`? Perhaps it is fine, but I am just wondering what kind of stability guarantees for this feature we are advertising by doing so?

Reply:
You're right: the usual way is to mark the field as EXPERIMENTAL (we have this mention for the `shards` field of the Prometheus CRD for instance). It isn't ideal but this is the best we can do because the other option would be to have a v2alpha CRD. 
Having said that, we could improve the situation a bit by using feature gates so that operator admins control exactly what users can/can't do (e.g. `--feature-gates=GlobalAlertmanagerConfig=true`).

Reply:
It was discussed in a previous contributors office hours meeting and the consensus was not to rely on feature gates (as long as the field is marked as EXPERIMENTAL).